### PR TITLE
Updated tests (to investigate test_lock_store.py behavior)

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- Updated tests to investigate `test_lock_store.py` behaviour
- Adjusted timeout for `test_cli_start` test

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: changelog/6735.misc.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Update tests to investigate ``test_lock_store.py`` behaviour`.
```
Comment (m-vdb) [inline changelog/6735.misc.rst:1]:
```suggestion
Update tests to investigate `test_lock_store.py` behavior`.
```

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -140,7 +140,6 @@ jobs:
         poetry-version: ${{ env.POETRY_VERSION }}
 
     - name: Set up virtual environment (Windows)
```
Comment (m-vdb) [inline .github/workflows/continous-integration.yml:142]:
```suggestion
    - name: Set up virtual environment
```

File: changelog/6735.misc.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Update tests to investigate ``test_lock_store.py`` behaviour`.
+
+Adjust a timeout for ``test_cli_start`` test.
```
Comment (m-vdb) [inline changelog/6735.misc.rst:3]:
```suggestion
Adjust a timeout for `test_cli_start` test.
```

Comment (m-vdb) [reply changelog/6735.misc.rst:3]: Oh actually this file should be markdown, not rst

**Review by m-vdb — CHANGES_REQUESTED** [review]

File: changelog/6735.misc.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Update tests to investigate `test_lock_store.py` behavior.
```
Comment (ricwo) [inline changelog/6735.misc.md:1]: I would add a sentence about how this has been updated. Something like 
```suggestion
Update tests in `test_lock_store.py`. Some of these tests are now allowed 
to fail on Windows and will be investigated further.
```

File: changelog/6735.misc.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Update tests to investigate `test_lock_store.py` behavior.
+
+Adjust a timeout for `test_cli_start` test.
```
Comment (ricwo) [inline changelog/6735.misc.md:3]:
```suggestion
Adjust the timeout for the `test_cli_start()` test in `tests/cli/test_cli.py` 
to allow for a slower start of the CLI on Windows.
```

File: tests/cli/test_cli.py [hunk 6]
```diff
@@ -16,7 +16,8 @@ def test_cli_start(run: Callable[..., RunResult]):
 
     duration = end - start
 
-    assert duration <= 7
+    # it sometimes takes a bit more time to start it on Windows
+    assert duration <= 10 if sys.platform == "win32" else 7
```
Comment (ricwo) [inline tests/cli/test_cli.py:20]: the docstring of this function needs to be updated as well

**Review by ricwo — APPROVED** [review]
👍

File: changelog/6735.misc.md (hunk 4 above)
Comment (alwx) [reply changelog/6735.misc.md:1]: It's not gonna be added to changelog anyway.

**Review by m-vdb — APPROVED** [review]

**Pull request merged**